multiversx_sdk/entrypoints/entrypoints.py

```diff
@@ -145,79 +142,61 @@ def create_network_provider(
-
+        self.chain_id = self._get_chain_id()
         return AccountTransactionsFactory(TransactionsFactoryConfig(self.chain_id))
 
     def create_relayed_controller(self) -> RelayedController:
-        if not self.chain_id:
-            self.chain_id = self.network_provider.get_network_config().chain_id
-
+        self.chain_id = self._get_chain_id()
         return RelayedController(self.chain_id)
 
     def create_relayed_transactions_factory(self) -> RelayedTransactionsFactory:
-        if not self.chain_id:
-            self.chain_id = self.network_provider.get_network_config().chain_id
-
+        self.chain_id = self._get_chain_id()
         return RelayedTransactionsFactory(TransactionsFactoryConfig(self.chain_id))
 
     def create_smart_contract_controller(self, abi: Optional[Abi] = None) -> SmartContractController:
-        if not self.chain_id:
-            self.chain_id = self.network_provider.get_network_config().chain_id
-
+        self.chain_id = self._get_chain_id()
         return SmartContractController(self.chain_id, self.network_provider, abi)
 
     def create_smart_contract_transactions_factory(self, abi: Optional[Abi] = None) -> SmartContractTransactionsFactory:
-        if not self.chain_id:
-            self.chain_id = self.network_provider.get_network_config().chain_id
-
+        self.chain_id = self._get_chain_id()
         return SmartContractTransactionsFactory(config=TransactionsFactoryConfig(self.chain_id), abi=abi)
 
     def create_token_management_controller(self) -> TokenManagementController:
-        if not self.chain_id:
-            self.chain_id = self.network_provider.get_network_config().chain_id
-
+        self.chain_id = self._get_chain_id()
         return TokenManagementController(self.chain_id, self.network_provider)
 
     def create_token_management_transactions_factory(
         self,
     ) -> TokenManagementTransactionsFactory:
-        if not self.chain_id:
-            self.chain_id = self.network_provider.get_network_config().chain_id
-
+        self.chain_id = self._get_chain_id()
         return TokenManagementTransactionsFactory(TransactionsFactoryConfig(self.chain_id))
 
     def create_transfers_controller(self) -> TransfersController:
-        if not self.chain_id:
-            self.chain_id = self.network_provider.get_network_config().chain_id
-
+        self.chain_id = self._get_chain_id()
         return TransfersController(self.chain_id)
 
     def create_transfers_transactions_factory(self) -> TransferTransactionsFactory:
-        if not self.chain_id:
-            self.chain_id = self.network_provider.get_network_config().chain_id
-
+        self.chain_id = self._get_chain_id()
         return TransferTransactionsFactory(TransactionsFactoryConfig(self.chain_id))
 
+    def _get_chain_id(self) -> str:
+        if self.chain_id:
+            return self.chain_id
+
+        return self.network_provider.get_network_config().chain_id
```

Comment:
Value should be cached in this function, instead of having the "caching" within each controller method. E.g.

```
def _acquire_chain_id() -> str:
    if missing, fetch & cache
    return cached

...

return TokenManagementController(self._acquire_chain_id(), self.network_provider)
```